sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/predicates.scala

```diff
@@ -138,46 +138,56 @@ case class Not(child: Expression)
 case class In(value: Expression, list: Seq[Expression]) extends Predicate {
 
   require(list != null, "list should not be null")
+
+  lazy val valExprs = value match {
+    case cns: CreateNamedStruct => cns.valExprs
+    case expr => Seq(expr)
+  }
+
+  override lazy val resolved: Boolean = {
+    lazy val checkForInSubquery = list match {
```

Comment:
Early evaluation causes failure like this: https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/80893/testReport/org.apache.spark.sql.hive.execution/HiveCompatibilitySuite/subquery_in_having/